.github/workflows/python-release.yaml

```diff
@@ -108,7 +108,7 @@ jobs:
 
 
   macos:
-    runs-on: macos-12-xl
+    runs-on: macos-14-xlarge
```

Comment:
cc @tychoish & @scsmithr you have context from #2543

Reply:
we should definitely just mirror whatever we're doing for the CLI.

my logic for choosing the 13 was just "use the oldest and smallest thing that works.